runtime-modules/content-directory/src/lib.rs

```
@@ -0,0 +1,1642 @@
+            let class_ids: Vec<T::ClassId> = <ClassById<T>>::enumerate().map(|(class_id, _)| class_id).collect();
+            for class_id in class_ids {
+                <ClassById<T>>::mutate(class_id, |class| {
+                    let class_permissions = class.get_permissions_mut();
+                    class_permissions.get_maintainers_mut().remove(&curator_group_id);
+                })
+            };
+            Ok(())
+        }
+
+        pub fn set_curator_group_status(
+            origin,
+            curator_group_id: T::CuratorGroupId,
+            is_active: bool,
+        ) -> dispatch::Result {
+            perform_lead_auth::<T>(origin)?;
+
+            Self::ensure_curator_group_exists(&curator_group_id)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            <CuratorGroupById<T>>::mutate(curator_group_id, |curator_group| {
+                curator_group.set_status(is_active)
+            });
+
+            Ok(())
+        }
+
+        /// Add curator to a given curator group
+        pub fn add_curator(
+            origin,
+            curator_group_id: T::CuratorGroupId,
+            curator_id: T::CuratorId,
+        ) -> dispatch::Result {
+            perform_lead_auth::<T>(origin)?;
+
+            Self::ensure_curator_group_exists(&curator_group_id)?;
+            Self::ensure_max_number_of_curators_limit_not_reached(curator_group_id)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            <CuratorGroupById<T>>::mutate(curator_group_id, |curator_group| {
+                curator_group.get_curators_mut().insert(curator_id);
+            });
+
+            Ok(())
+        }
+
+        /// Remove curator from a given curator group
+        pub fn remove_curator(
+            origin,
+            curator_group_id: T::CuratorGroupId,
+            curator_id: T::CuratorId,
+        ) -> dispatch::Result {
+            perform_lead_auth::<T>(origin)?;
```

I think this should fail loudly, hence it should check that curator actually is part of group.

---

Good point, fixed it here https://github.com/iorveth/substrate-runtime-joystream/commit/09db913000644f1774554df61aed1af71de425f0